content/articles/2018-10-devtools-2.0.0.Rmarkdown

```diff
@@ -0,0 +1,256 @@
+spelling release post](https://ropensci.org/technotes/2017/09/07/spelling-release/).
+
+## Acknowledgements
+
+This release was truly a team effort! Much of the work in the uncoupled
+packages, particularly the remotes, rcmdcheck, revdepcheck and sessioninfo
+packages was done by Gábor Csárdi. Hadley Wickham worked extensively on
+pkgbuild, pkgload and usethis (as well as being responsible for most of the
+original code in devtools) and Jenny Bryan had major contributions to the
+usethis package.
+
+```{r include = F, eval = F}
+library(tidyverse)
+pkgs <- set_names(c("remotes", "rcmdcheck", "revdepcheck", "sessioninfo", "pkgbuild", "pkgload", "usethis"))
+
+issues <- map(pkgs, ~ gh::gh("GET /repos/:owner/:repo/issues", owner = "r-lib", repo = .x, .limit = Inf, state = "closed"))
+
+# 2017-05-08 is the date of the last major devtools release 1.13.0
+issues$devtools <- gh::gh("GET /repos/:owner/:repo/issues", owner = "r-lib", repo = "devtools", .limit = Inf, since = "2017-05-08", state = "closed")
+
+# We need to filter only issues closed after the date
+issues$devtools <- keep(issues$devtools, ~ as.POSIXct(.x$closed_at) > "2017-05-08")
+
+lengths(issues)
+
+# Closed issues
+sum(lengths(issues))
+#> 1487
+```
+
+```{r include = FALSE, eval = F}
+
+contributors <- map(c(pkgs, devtools = "devtools"), ~ gh::gh("GET /repos/:owner/:repo/stats/contributors", owner = "r-lib", repo = .x, .limit = Inf))
+
+tally_weeks <- function(x) {
+  map_dfr(x, ~ list(additions = .x$a, deletions = .x$d, commits = .x$c)) %>%
+    summarise_all(sum)
+}
+
+tally_contributor <- function(x) {
+  user <- x$author$login
+  weeks <- keep(x$weeks, ~ .POSIXct(.x$w) > "2017-05-08")
+  tally_weeks(weeks) %>%
+    dplyr::filter(commits > 0) %>%
+    mutate(user = user)
+}
+
+contributions <- map_dfr(contributors, ~ map_dfr(.x, tally_contributor), .id = "package")
+
+contributions %>% count(user, wt = commits) %>% arrange(desc(n))
+
+contributions %>% dplyr::filter(user == "yiufung")
+
+contributions %>% count(wt = commits)
+#> 1579
+
+# Not sure how meaningful this is, as it includes things like pkgdown sites :/
+contributions %>% count(wt = additions + deletions)
+#> 155253
+
+contributions %>% count(user) %>% pull("n") %>% sum()
+```
+
+The work spanned over 8 packages (devtools + the uncoupled packages) and
+includes 1,579 commits, 1,487 closed issues and 107 different code contributors!
+
+```{r include = FALSE, eval = FALSE}
+issue_contributors <- unique(unlist(map(str_glue("r-lib/{c(pkgs, 'devtools')}"), usethis::use_tidy_thanks, from = "2017-05-08")))
+
+# Remove bots
+issue_contributors <- grep("bot", issue_contributors, value = TRUE, invert = TRUE)
+
+str_glue("[@&#8203;{issue_contributors}](https://github.com/{issue_contributors})") %>% glue::glue_collapse(sep = ", ", last = ", and ") %>% clipr::write_clip()
+```
+
+We are of course grateful to _all_ of the *391* people who contributed not just code, but also issues and comments for this release:
```

Comment:
I'm getting 145 from this list.

Reply:
I recalculated it, there should be 336 contributors now, I think the 391 was copied from a previous ggplot post and got inadvertently reverted in an undo session gone awry.